src/components/cl/hier/cl_hier.h

```diff
@@ -63,7 +63,10 @@ typedef struct ucc_cl_hier_context_config {
 typedef struct ucc_cl_hier_lib {
     ucc_cl_lib_t             super;
     ucc_cl_hier_lib_config_t cfg;
-    ucc_config_names_array_t tls; /*< Intersection of UCC_CL_HIER_TLS vs sbgp_tls */
+    ucc_config_allow_list_t  tls; /*< Intersection of UCC_CL_HIER_TLS
+                                    vs sbgp_tls */
+    ucc_config_names_array_t tls_forced; /*< set of TLs that are
```

Comment:
seems like tls_force will be part of each CL, then it makes sense to add it to base class

Reply:
moved tls_forced field to common super class. The cleanup is also made in base destructor. Initialization is still CL specific